applications/appunorganized/appunorganized-plugins/org.csstudio.swt.rtplot/src/org/csstudio/swt/rtplot/internal/TracePainter.java

```diff
@@ -238,6 +242,10 @@ final private void drawValueLines(final GC gc,
             if (Double.isNaN(value))
             {
                 flushPolyLine(gc, value_poly, line_width);
+                if (last_y != -1)
```

Comment:
These checks may be at issue. We discussed yesterday that they are not expected to be needed. I will try removing them.